docs/DEVELOPING.md

```diff
@@ -717,6 +717,188 @@ $ make ftest NAME=mssql
 
 ℹ️ The e2e test uses default values defined in [Configure Microsoft SQL connector](#configure-microsoft-sql-connector)
 
+
+## SharePoint Connector
+
+The [Elastic SharePoint connector](../connectors/sources/sharepoint.py) is provided in the Elastic connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
+
+### Availability and prerequisites
+
+⚠️ _Currently, this connector is available in **beta** starting in 8.8_.
+Features in beta are subject to change and are not covered by the service level agreement (SLA) of features that have reached general availability (GA).
+
+Elastic versions 8.6.0+ are compatible with Elastic connector frameworks. Your deployment must include the Elasticsearch, Kibana, and Enterprise Search services.
+
+### Compatibility
+
+  Sharepoint Connector supports below Sharepoint Server versions:
+  - Sharepoint 2013
+  - Sharepoint 2016
+  - Sharepoint 2019
+
+
+### Create an Administrator Account For SharePoint Online:
+  - To create an administrator account, join the Microsoft 365 Developer Program and get the Microsoft 365 developer subscription with a new administrator account for 90 days. Use this administrator account for signing into SharePoint Online and Azure AD to fetch more details.
+  - Register your app with Microsoft Azure AD
+  - To use the Microsoft identity platform endpoint, you must register your app using the Azure app registration portal. You can use either a Microsoft account or a work or school account to register an app.
```

Comment:
We need to give actual instructions on how to do this. See the old Workplace Search docs: https://www.elastic.co/guide/en/workplace-search/current/workplace-search-sharepoint-online-connector.html#sharepoint-configuration